internal/services/v1/relationships.go

```diff
@@ -148,12 +149,47 @@ func (ps *permissionServer) ReadRelationships(req *v1.ReadRelationshipsRequest,
 		DispatchCount: 1,
 	})
 
+	limit := 0
+	var startCursor options.Cursor
+
+	rrRequestHash, err := computeReadRelationshipsRequestHash(req)
+	if err != nil {
+		return shared.RewriteError(ctx, err)
+	}
+
+	if req.OptionalCursor != nil {
+		decodedCursor, err := cursor.DecodeToDispatchCursor(req.OptionalCursor, rrRequestHash)
+		if err != nil {
+			return shared.RewriteError(ctx, err)
+		}
+
+		if len(decodedCursor.Sections) != 1 {
+			return shared.RewriteError(ctx, NewInvalidCursorErr("did not find expected resume relationship"))
+		}
+
+		parsed := tuple.Parse(decodedCursor.Sections[0])
+		if parsed == nil {
+			return shared.RewriteError(ctx, NewInvalidCursorErr("could not parse resume relationship"))
+		}
```

Comment:
maybe turn this into a helper like "ExactlyOneSection" which returns an error if it does not

Reply:
Same idea